lifecycle.go

```diff
@@ -198,6 +198,31 @@ var defaultLogConsumersHook = func(cfg *LogConsumerConfig) ContainerLifecycleHoo
 	}
 }
 
+func checkPortsMapped(exposedAndMappedPorts nat.PortMap, exposedPorts []string) error {
+	portMap, _, err := nat.ParsePortSpecs(exposedPorts)
+	if err != nil {
+		return err
+	}
+
+	for exposedPort, _ := range portMap {
+		// having entries in exposedAndMappedPorts, where the key is the exposed port,
+		// and the value is the mapped port, means that the port has been already mapped.
+		if _, ok := exposedAndMappedPorts[exposedPort]; !ok {
+			// check if the port is mapped with the protocol (default is TCP)
+			if !strings.Contains(string(exposedPort), "/") {
+				exposedPort = nat.Port(fmt.Sprintf("%s/tcp", string(exposedPort)))
+				if _, ok := exposedAndMappedPorts[exposedPort]; !ok {
+					return fmt.Errorf("port %s is not mapped yet", exposedPort)
+				}
+			} else {
+				return fmt.Errorf("port %s is not mapped yet", exposedPort)
+			}
```

Comment:
suggestion: Using early return reduces indentation and makes it easier to grok.
The creation of the port can also be simplified eliminating the `fmt.Sprintf`
```suggestion
			if strings.Contains(string(exposedPort), "/") {
				return fmt.Errorf("port %s is not mapped yet", exposedPort)
			}
			
			// Port didn't have a type, default to tcp and retry.
			exposedPort = nat.Port(exposedPort +"/tcp")
			if _, ok := exposedAndMappedPorts[exposedPort]; !ok {
				return fmt.Errorf("port %s is not mapped yet", exposedPort)
			}
```

Reply:
https://github.com/testcontainers/testcontainers-go/pull/2658/commits/5fe2f77b1c6101b2c550f182ec4021d828ff5bd0